polars/polars-utils/src/sort.rs

```diff
@@ -40,3 +41,40 @@ pub unsafe fn perfect_sort(pool: &ThreadPool, idx: &[(IdxSize, IdxSize)], out: &
     // all elements are written
     out.set_len(idx.len());
 }
+
+macro_rules! perfect_sort_inner {
```

Comment:
Why this macro? I only see it used ones and code in functions is easier to read/maintain than macros.